src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -107,49 +115,55 @@ class Task extends React.Component<Props, State> {
         this.setState({ modalError: error });
     };
 
-    fetchTaskCollaborators = (): Promise<any> => {
-        const { id, api, task_links } = this.props;
-        const { entries } = task_links;
-        let fileId = '';
+    getAllTaskCollaborators = (onSuccess: () => any) => {
+        const { id, api, task_links, assigned_to } = this.props;
+        const { errorOccured, taskCollaboratorLoadErrorMessage } = messages;
 
-        if (entries[0] && entries[0].target) {
-            fileId = entries[0].target.id;
+        // skip fetch when there are no additional collaborators
+        if (!assigned_to.next_marker) {
+            this.setState({ assignedToFull: assigned_to });
+            onSuccess();
+            return;
         }
 
+        // fileid is required for api calls, check for presence
+        const fileId = get(task_links, 'entries[0].target.id');
         if (!fileId) {
-            return Promise.reject();
+            return;
         }
 
         this.setState({ isLoading: true });
+        api.getTaskCollaboratorsAPI(false).getTaskCollaborators({
```

Comment:
This is likely outside the scope of this PR, but it seems like all the data access and formatting should be happening in some kind of API module or container component rather than being mixed into the UI component.

Reply:
I was thinking the same for this one, but wasn't sure where

Reply:
I'm sure there are lots of options.  One option would be to add a wrapping container component, maybe called TaskContainer, that only handles data access and state management. You can see an example of this pattern with the [VersionsSidebarContainer](https://github.com/box/box-ui-elements/blob/master/src/elements/content-sidebar/versions/VersionsSidebarContainer.js) and its child UI component [VersionsSidebar](https://github.com/box/box-ui-elements/blob/master/src/elements/content-sidebar/versions/VersionsSidebar.js). The rule of thumb I follow is that declared HTML should live in a separate component from data access.

Reply:
We can do this as part of refactor bucket.